dev_setup.sh

```diff
@@ -321,7 +321,7 @@ function install_venv() {
     # verified functional.
     curl https://bootstrap.pypa.io/3.3/get-pip.py | "${VIRTUALENV_ROOT}/bin/python" - 'pip==18.0.0'
     # Function status depending on if pip exists
-    [ -x "${VIRTUALENV_ROOT}/bin/pip" ]
+    [[ -x ${VIRTUALENV_ROOT}/bin/pip ]]
```

Comment:
Could this fail if there's spaces in the path of the VIRTUAENV_ROOT?

Reply:
@forslund 
Nope, this is only an issue with single square brackets.
- `[` is a command, which requires strings to compare/check as single arguments, thus strings with spaces must be quoted.
- `[[` is bash syntax, which takes the string as is with spaces. Only quotations are required when doing string comparisons on the right side string, if you don't want globs to be expanded. On the right side otherwise e.g. `[01]` or `*` are handled as globs/wildcards to compare. But even there, spaces are no issue. When doing file checks, nothing needs to be taken care of, besides when `$` of quote characters should be taken literally of course.

Little test:
```
2019-06-18 19:19:26 root@micha:/tmp# mkdir 'test path with spaces'
2019-06-18 19:19:53 root@micha:/tmp# ls -Al
...
drwxr-xr-x 2 root root   40 Jun 18 19:19 'test path with spaces'
2019-06-18 19:19:54 root@micha:/tmp# dir='test path with spaces'
2019-06-18 19:20:08 root@micha:/tmp# [[ -d $dir ]] && echo yes
yes
2019-06-18 19:20:18 root@micha:/tmp# > 'test path with spaces'/testfile
2019-06-18 19:20:33 root@micha:/tmp# [[ -f $dir/testfile ]] && echo yes
yes
```

But with single square braces:
```
2019-06-18 19:20:52 root@micha:/tmp# [ -f $dir/testfile ] && echo yes
bash: [: too many arguments
```

Reply:
Awesome, that's really good to know. I love beig corrected like this :)

Reply:
@forslund 
Always good to be opened for learning/being "corrected". However this is just a minor non-measurable thing, shipping some consistency, where I couldn't hold myself as a perfectionist when checking the code 😄.

Awesome the have an open source AI assistant!